recipes/blast/build.sh

```diff
@@ -1,26 +1,59 @@
 #!/bin/bash
 
-set -e -x -o pipefail
+cd $SRC_DIR/c++/
 
-mkdir -p $PREFIX/bin
+export CFLAGS="$CFLAGS -O2"
+export CXXFLAGS="$CXXFLAGS -O2"
+export CPPFLAGS="$CPPFLAGS -I$PREFIX/include"
+export LDFLAGS="$LDFLAGS -L$PREFIX/lib"
 
-if [ "$(uname)" == "Darwin" ]; then
-    echo "Platform: Mac"
-
-    cp $SRC_DIR/bin/* $PREFIX/bin/
-
-elif [ "$(expr substr $(uname -s) 1 5)" == "Linux" ]; then
-    echo "Platform: Linux"
+if test x"`uname`" = x"Linux"; then
+    # only add things needed; not supported by OSX ld
+    LDFLAGS="$LDFLAGS -Wl,-as-needed"
+fi
 
-    cd $SRC_DIR/c++/
-    # --with-hard-runpath is needed otherwise BLAST programs would search
-    # libraries first in the directories defined by the LD_LIBRARY_PATH
-    # environment variable, instead of using the rpath specified by conda
-    ./configure --prefix=$PREFIX --with-hard-runpath
```

Comment:
If you don't mind waiting a bit, I'd like to test if this is really not needed any more and maybe add a test to prevent future regressions.

For reference, this was needed because on my cluster the `$LD_LIBRARY_PATH` environment variable is set and blast programs were picking up libstdc++.so.6 from there, resulting in errors like:
```
makeblastdb: /PATH/IN/MY/LD_LIBRARY_PATH/libstdc++.so.6: version `GLIBCXX_3.4.15' not found (required by makeblastdb)
```

Reply:
As I was worried, the test case I've added fails without `--with-hard-runpath`, I'm going to re-add it, test it and push here, if all goes well.

Reply:
I had already changed it to `--with-runpath` ...

Reply:
Ugh. I don't know about this. That sounds like a broken cluster setup. A better way to get blast to ignore LD_LIBRARY_PATH is to remove the broken setting: `unset LD_LIBRARY_PATH; export LD_LIBRARY_PATH`.